---
title: Python Work
author: Taimur Hassan
date: '2020-05-14'
slug: '{python}'
categories: []
tags: []
description: ''
---

![](python.png)

In bioinformatics, k-mers refer to all the possible subsequences (of length k) from a read obtained through DNA sequencing. For example, if the DNA sequencing read is "ATCATCATG", then a the 3-mers in that read include "ATC" (which occurs twice), "TCA" (which occurs twice), "CAT" (occurs twice), and "ATG" (occurs once). You can read more about k-mers on Wikipedia.

Here is a function that takes a string of nucleotides as input and returns a dictionary with all 3-mers present in that string, and the number of times that each 3-mer occurs. Then this function can be used to find the 3-mers in the DNA sequence my_seq1 and my_seq2 defined below.

The output of my function will be a dictionary that is structured like this (although it will have several more entries)

{"CAG": 2, "AGC": 1, "GCC": 3}

where each key is a 3-mer itself (e.g., "ATC") and each value is the number of times that 3-mer occurs.




CODE:
```{r}
library(reticulate)
use_virtualenv("r-reticulate")
```

```{python}
my_seq1 = "ATCATCATG"
my_seq2 = "CAGCCCAATCAGGCTCTACTGCCACTAAACTTACGCAGGATATATTTACGCCGACGTACT"

def hw11(seq):
    out_dict={}
    for i in range(len(seq) - 2):
        if seq[i:i+3] in out_dict:
            out_dict[seq[i:i+3]] +=1
        else:
            out_dict[seq[i:i+3]]=1
    return out_dict
print(hw11(my_seq1))

def hw12(seq):
    out_dict={}
    for i in range(len(seq) - 2):
        if seq[i:i+3] in out_dict:
            out_dict[seq[i:i+3]] +=1
        else:
            out_dict[seq[i:i+3]]=1
    return out_dict
print(hw12(my_seq2))


Plist= [11,22,33,44,55,66]

print(Plist)


def Psq_fun (x):
  value= x*x 
  return(value)
  
print(Psq_fun(9))


```


